unit_tests/log_transfer/test_main_chkpt_info.cpp

```diff
@@ -0,0 +1,549 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#define CATCH_CONFIG_MAIN
+#include "catch2/catch.hpp"
+
+#include "log_checkpoint_info.hpp"
+#include "log_lsa.hpp"
+#include "log_record.hpp"
+
+#include <algorithm>
+#include <vector>
+#include <random>
+#include <cstdlib>
+
+using namespace cublog;
+
+
+class test_env_chkpt
+{
+  public:
+    test_env_chkpt ();
+    test_env_chkpt (int size_trans, int size_sysops);
+    ~test_env_chkpt ();
+
+    LOG_LSA generate_log_lsa();
+    LOG_INFO_CHKPT_TRANS generate_log_info_chkpt_trans();
+    LOG_INFO_CHKPT_SYSOP generate_log_info_chkpt_sysop();
+    std::vector<LOG_LSA> used_logs;
+
+    const int max = 32700;
+    static void require_equal (checkpoint_info before, checkpoint_info after);
+
+    checkpoint_info before;
+    checkpoint_info after;
+  private:
+
+};
+
+TEST_CASE ("Test pack/unpack checkpoint_info class 1", "")
+{
+  test_env_chkpt env {100, 100};
+
+  cubpacking::packer pac;
+  size_t size = env.before.get_packed_size (pac, 0);
+  char buffer[size];
+  pac.set_buffer (buffer, size);
+  env.before.pack (pac);
+
+  cubpacking::unpacker unpac;
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+
+  env.require_equal (env.before, env.after);
+}
+
+TEST_CASE ("Test pack/unpack checkpoint_info class 2", "")
+{
+  test_env_chkpt env {0, 100};
+
+  cubpacking::packer pac;
+  size_t size = env.before.get_packed_size (pac, 0);
+  char buffer[size];
+  pac.set_buffer (buffer, size);
+  env.before.pack (pac);
+
+  cubpacking::unpacker unpac;
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+
+  env.require_equal (env.before, env.after);
+}
+
+TEST_CASE ("Test pack/unpack checkpoint_info class 3", "")
+{
+  test_env_chkpt env {220, 80};
+
+  cubpacking::packer pac;
+  size_t size = env.before.get_packed_size (pac, 0);
+  size += env.before.get_packed_size (pac, size);
+  char buffer[size];
+  pac.set_buffer (buffer, size);
+  env.before.pack (pac);
+  env.before.pack (pac);
+
+  cubpacking::unpacker unpac;
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+  env.require_equal (env.before, env.after);
+
+  checkpoint_info after_2;
+
+  after_2.unpack (unpac);
+
+  env.require_equal (env.after, after_2);
+
+}
+
+test_env_chkpt::test_env_chkpt () : test_env_chkpt (100, 100)
+{
+}
+
+test_env_chkpt::test_env_chkpt (int size_trans, int size_sysops)
+{
+  srand (time (0));
+
+  log_lsa log_to_add = this->generate_log_lsa();
+  before.m_start_redo_lsa = log_to_add;
```

Comment:
Write the code in the easiest way possible:
```suggestion
  before.m_start_redo_lsa = this->generate_log_lsa();
```